spec/requests/objects_controller_primary_moab_spec.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+RSpec.describe ObjectsController, type: :request do
+  let(:pres_obj) { create(:preserved_object) }
+  let(:primary_moab) { create(:complete_moab) }
+
+  describe 'GET #primary_moab' do
+    context 'when object found' do
+      before do
+        allow(pres_obj.preserved_objects_primary_moab).to receive(:complete_moab).and_return(primary_moab)
```

Comment:
we used this pattern for testing a bunch when we first started working on pres cat, but i think our experience has been that setting up db objects using FactoryBot has been less brittle, because it ties less tightly to specific query implementations and how specifically objects get loaded, allowing for easier refactoring later on.

what about something like:
```ruby
# on what's currently lines 5 and 6, which would remove the need for this before block i think?
let(:primary_moab) do
  create(:complete_moab).tap { |cm| PreservedObjectsPrimaryMoab.create(complete_moab: cm, preserved_object: cm.preserved_object) }
end
let(:pres_obj) { primary_moab.preserved_object }
```

and then, with the controller change suggested above, the rest of test code could be left pretty much as-is, other than changing the success expectations to be something like `expect(response.body).to eq(primary_moab.moab_storage_root.storage_location)` (depending on whether or not you preferred plain-text or JSON for the response, or whatever).